src/views/pages/ManagePeople/index.vue

```diff
@@ -52,81 +54,68 @@ export default Vue.extend({
       this.searched = searchByName(this.people, this.search);
     },
   },
-  created: function created() {
-    this.searched = this.people;
+  created: async function created() {
+    const self = this;
+
+    // get people types
+    const personCategories: any[] = [];
+    await axios.get('http://boresha.tech:9090/people/categories')
+      .then((response: any) => {
+        response.data.map((category: any) => {
+          personCategories.push(category.name);
+        });
+      })
+      .catch((error: any) => {
+        console.log(error.message);
+        self.error = error.message;
+      });
+
+    // get all people
+    personCategories.map((category: string) => {
+      // get all people of particular category
+      axios.get('http://boresha.tech:9090/people/' + category)
+        .then((response: any) => {
+          // construct each person
+          response.data.map((person: any) => {
+            // construct full name
+            const fullName: string =
+              `${person.firstName || ''} ${person.middleName || ''} ${person.lastName || ''}`;
+
+            // construct phone number
+            let fullPhone: string = '';
+            if (person.phoneCountry) {
+              fullPhone += `+${person.phoneCountry}`;
+            }
+            if (person.phoneArea) {
+              fullPhone += ` (${person.phoneArea})`;
+            }
+            if (person.phoneNumber) {
+              fullPhone += ` ${person.phoneNumber.slice(0, 3)}-${person.phoneNumber.slice(3)}`;
+            }
+
+            self.people.push({
+              name: fullName,
+              phoneNumber: fullPhone,
+              peopleCategory: person.peopleCategory,
+              lastModified: new Date(person.lastModified).toUTCString(),
+            });
+          });
+        })
+        .catch((error: any) => {
+          console.log(error.message);
+          self.error = error.message;
+        });
+    });
+
+    // update list
+    self.searched = self.people;
```

Comment:
This has not remedied the issue we spoke about last time. Here you are mutating data in a closure inside of map, and then not waiting for it to complete. If you wanted this to work correctly you have to await all of the promises returned by `map`. Ideally it would be better to structure this in a way so that the data you want is returned by the `map`, and only use pure functions in the map callback. That is easier to reason.